internal/storage/mock_control_client.go

```diff
@@ -43,3 +43,15 @@ func (m *MockStorageControlClient) DeleteFolder(ctx context.Context,
 	args := m.Called(ctx, req, opts)
 	return args.Error(0)
 }
+
+func (m *MockStorageControlClient) GetFolder(ctx context.Context,
+	req *controlpb.GetFolderRequest,
+	opts ...gax.CallOption) (*controlpb.Folder, error) {
+	args := m.Called(ctx, req, opts)
+
+	if folder, ok := args.Get(0).(*controlpb.Folder); ok {
```

Comment:
Why are we writing this logic here? Shouldn't we just return whatever we get from the main method? Otherwise, we will not be able to properly check if the error is nil in the case of a successful get object.

Reply:
This is needed, as in positive scenario error is not returned , mock method should also return nil and in folder not found scenario, error is returned, but folder is not returned, hence single line cannot assert both

Reply:
May be we can write same comment so in future we have reference why we did?